### PR TITLE
Restructure NIN project

### PR DESCRIPTION
**Pull request by aleksanb**
This restructuring simplifies the publishing process as everything is located under a single package.json as opposed to the previous structure of two package.jsons on the same level, requiring us to copy files over for publishing.

### CODE REVIEW

**Comment by sigvef** [comment]
We're not going with the project structure that we discussed?

```
nin/
├── backend/
│   ├── package.json
├── dasBoot/
├── frontend/
│   ├── package.json
└── package.json
```

**Comment by aleksanb** [comment]
I tried that at first, but with the rationale that i didn't want to rewrite all references to __dirname as i thought __dirname always was the directory of the executable ran, i unrolled all files under nin/.
As it turns out that __dirname is module-local (https://nodejs.org/docs/latest/api/globals.html#globals_dirname) i can switch to the discussed structure instead.

**Comment by aleksanb** [comment]
Updated to reflect the all-new all-old structure!

File: nin/backend/serve.js [hunk 1]
```diff
@@ -10,17 +10,9 @@ var readDir = require('readdir');
 var socket = require('./socket');
 var watch = require('./watch');
 
-var publishing = false;
-
 
 var serve = function(projectPath, shouldRunHeadlessly) {
-
-  var genPath = p.join(projectPath, '/gen/');
-  mkdirp.sync(genPath);
```
Comment (sigvef) [inline nin/backend/serve.js]: This used to create the `gen/` folder in a demo project's folder. We probably still want to do this, as other parts of nin assume that this directory exists.

**Comment by aleksanb** [comment]
Addressed, it worked locally due to me having created gen from an earlier run of nin.

**Comment by stianjensen** [comment]
How does this work when published? npm handles installing dependencies from packge.json in nested directories?

**Comment by aleksanb** [comment]
No idea how it works when published.

**Comment by stianjensen** [comment]
I guess it would work if the dependencies listed in backend/package.json are moved out to the top-most package.json

**Comment by aleksanb** [comment]
I'll do that then.

**Comment by aleksanb** [comment]
It's done.
Publish through make publish, as that will build the frontend project so that it can be added during the publishing step.

**Pull request merged**